src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.cpp

```diff
@@ -939,7 +939,7 @@ void ControlBoardWrapper::run()
 //
 bool ControlBoardWrapper::setPid(int j, const Pid &p)
 {
-    int off=device.lut[j].offset;
+    int off; try{off = device.lut.at(j).offset;} catch(...){yError() << "joint number " << j <<  " out of bound [0-"<< controlledJoints << "] for part " << partName; return false; }
```

Comment:
My point is that it can be equal or better than if, not worse. Then how much depends on the implementation of `.at()` vs `if` and compilers.
This is not a matter of profiling nor optimization, the question was " why `try` instead of `if`?" The answer is that in this case it can only be equal or better.